### PR TITLE
TGP-1649: Split up EIS config into HAWK and PEGA

### PR DESCRIPTION
**Pull request by IoannisLazaridis-oc**
### Description
- [x] Split the EIS configuration into separate HAWK and PEGA paths.
- [x] Updated the `BaseConnector` to handle  both HAWK and PEGA Instances.
- [x] Modified the `BaseIntegrationWithConnectorSpec `so we can test both connectors on the same test like `RequestAdviceIntegrationSpec`.
- [x] Updated all the integration test to reflect the changes.

### CODE REVIEW

File: conf/application.conf [hunk 1]
```diff
@@ -93,25 +93,33 @@ microservice {
       host = localhost
       port = 8500
     }
-    eis {
+    hawk {
         protocol = http
         host = localhost
-        port = 10903
+        port = 10908
```
Comment (taylors-version) [inline conf/application.conf:99]: Leave this as 10903 so we use the static stubs by default

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/BaseConnector.scala [hunk 2]
```diff
@@ -28,26 +28,26 @@ trait BaseConnector {
...
   def dateTimeService: DateTimeService
 
-  def buildHeaders(correlationId: String, accessToken: String)(implicit hc: HeaderCarrier): Seq[(String, String)] =
+  protected def buildHeaders(correlationId: String, accessToken: String, forwardedHost: String)(implicit
+    hc: HeaderCarrier
+  ): Seq[(String, String)] =
     Seq(
       HeaderNames.CorrelationId -> correlationId,
-      HeaderNames.ForwardedHost -> appConfig.eisConfig.forwardedHost,
+      HeaderNames.ForwardedHost -> forwardedHost,
```
Comment (AmarkumarJagtap) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/BaseConnector.scala:36]: think forwarded host is same for both, may be pass directly from config (hawk and pega) if its different then we can remove appConfig from  declaration at line 28

File: conf/application.conf (hunk 1 above)
Comment (IoannisLazaridis-oc) [reply conf/application.conf]: Done ✅

**Review by taylors-version — APPROVED** [review]

**Review by pauloncast — APPROVED** [review]

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/BaseConnector.scala (hunk 2 above)
Comment (taylors-version) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/BaseConnector.scala:36]: Whilst I can't see B&T and Case Management needing separate ones in the future, I'm happy for this to remain as two distinct configs for now

**Review by AmarkumarJagtap — APPROVED** [review]

**Pull request merged**